strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/services/impl/CronTaskConfigurationServiceImpl.java

```diff
@@ -1,92 +1,93 @@
 package org.carlspring.strongbox.cron.services.impl;
 
+import javax.inject.Inject;
+
 import org.carlspring.strongbox.cron.domain.CronTaskConfigurationDto;
 import org.carlspring.strongbox.cron.domain.CronTasksConfigurationDto;
-import org.carlspring.strongbox.cron.domain.GroovyScriptNamesDto;
 import org.carlspring.strongbox.cron.exceptions.CronTaskException;
 import org.carlspring.strongbox.cron.exceptions.CronTaskNotFoundException;
 import org.carlspring.strongbox.cron.jobs.AbstractCronJob;
 import org.carlspring.strongbox.cron.services.CronJobSchedulerService;
 import org.carlspring.strongbox.cron.services.CronTaskConfigurationService;
 import org.carlspring.strongbox.cron.services.CronTaskDataService;
 import org.carlspring.strongbox.event.cron.CronTaskEventListenerRegistry;
-
-import javax.inject.Inject;
-
 import org.quartz.SchedulerException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
-import org.springframework.context.ApplicationContext;
-import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextStartedEvent;
 import org.springframework.stereotype.Service;
 
 @Service
 class CronTaskConfigurationServiceImpl
-        implements CronTaskConfigurationService, ApplicationContextAware
+        implements CronTaskConfigurationService, ApplicationListener<ContextStartedEvent>
 {
 
     private final Logger logger = LoggerFactory.getLogger(CronTaskConfigurationServiceImpl.class);
 
     @Inject
     protected CronTaskEventListenerRegistry cronTaskEventListenerRegistry;
 
-    private AutowireCapableBeanFactory beanFactory;
-
     @Inject
     private CronTaskDataService cronTaskDataService;
 
     @Inject
     private CronJobSchedulerService cronJobSchedulerService;
 
     @Override
-    public void setApplicationContext(final ApplicationContext context)
+    public void onApplicationEvent(ContextStartedEvent event)
     {
-        beanFactory = context.getAutowireCapableBeanFactory();
+        CronTasksConfigurationDto cronTasksConfiguration = getTasksConfigurationDto();
+
+        for (CronTaskConfigurationDto dto : cronTasksConfiguration.getCronTaskConfigurations())
+        {
+            if (dto.isOneTimeExecution())
+            {
+                continue;
+            }
+            
+            cronJobSchedulerService.scheduleJob(dto);
+            
+            if (dto.shouldExecuteImmediately())
+            {
+                cronJobSchedulerService.executeJob(dto);
+            }
+        }
     }
 
     public void saveConfiguration(CronTaskConfigurationDto configuration)
-            throws Exception
```

Comment:
Who keeps adding these `throws Exception` statements? It looks like an IDE mis-configuration. I see it all over tests.

Thanks for clearing it out here!